cryptographic_estimators/MAYOEstimator/MAYOAlgorithms/direct_attack.py

```diff
@@ -78,6 +80,29 @@ def __init__(self, problem: MAYOProblem, **kwargs):
         
         self._fastest_algorithm = None
 
+    @optimal_parameter
+    def K(self):
+        """
+        Return the optimal parameter `K` from Furue, Nakamura, and Takagi strategy
+
+        EXAMPLES::
+
+            sage: from cryptographic_estimators.MAYOEstimator.MAYOAlgorithms.direct_attack import DirectAttack
+            sage: from cryptographic_estimators.MAYOEstimator.mayo_problem import MAYOProblem
+            sage: E = DirectAttack(MAYOProblem(n=80, m=60, o=18, k=12, q=16))
+            sage: E.K()
+            15
+                
+        """
+        if self._optimal_parameters.get("K") is None:
+            n, m, _, k, q = self.problem.get_parameters()
+            if self.complexity_type == ComplexityType.ESTIMATE.value:
+                return _optimize_k(n=k*n, m=m, k=k, q=q)
```

Comment:
this doesn't follow the general design by implementing the optimization routine outside the framework.

Reply:
if there is a strong argument to do it, please specify, otherwise we should adhere to the usual framework I think

Reply:
Agree, it doesn't follow the general optimization routine. 
The strong argument to do it is efficiency. 
In the general framework we would need to use the `_compute_time_complexity` as the function to optimize, which in this case is the MQEstimator.fastest_algorithm function, and it is super slow. 
In addition, the MAYO estimator uses BooleanSolveXL as the optimization function so we do so. Still we use the MQEstimator.fastest_algorithm  to get time complexity estimates using the optimal `K` computed with BooleanSolveXL.

Reply:
Ok, good for me. I think it's not even outside the framework. It seems the K parameter can be optimized independently. In those cases we don't need to use the `get_optimal_parameters` (see eg. the `r` parameter in the `SDAlgorithm` class).

Reply:
At the beginning we implemented it as `r` in `SDAlgorithm` but then we realized this was not efficient since `K` is only used for the `direct_attack` (opposite as `r`). Hence, no need  to optimized when you want to estimate other attack, for instance, `kipnis_shamir`.

@fmerino21 could you confirm if this is correct?

Reply:
Exactly as Javier said, it is not efficient to compute `K` for all algorithms given that we are only using it in the `direct_attack`. Besides that, by implementing the `K` parameter that way, when we use the method `.table(show_all_parameters=1)` in an instance of the `MAYOEstimator`, the `K` parameter is shown in all the algorithms, which makes no sense.